test/e2e_node/runner/remote/run_remote.go

```
@@ -497,24 +493,24 @@ func getGCEImages(imageRegex, imageFamily string, project string, previousImages
 					creationTime: creationTime,
 					name:         instance.Name,
 				}
-				klog.V(4).Infof("Found image %q based on regex %q and family %q in project %q", io.string(), imageRegex, imageFamily, project)
 				imageObjs = append(imageObjs, io)
 			}
 			return nil
 		},
 	); err != nil {
 		return nil, fmt.Errorf("failed to list images in project %q: %v", project, err)
 	}
+
+	// Pick the latest image after sorting.
 	sort.Sort(byCreationTime(imageObjs))
 	images := []string{}
-	for _, imageObj := range imageObjs {
-		images = append(images, imageObj.name)
-		previousImages--
-		if previousImages < 0 {
-			break
-		}
+	if len(imageObjs) > 0 {
+		klog.V(4).Infof("found images %+v based on regex %q and family %q in project %q", imageObjs, imageRegex, imageFamily, project)
+		images = append(images, imageObjs[0].name)
+		return images, nil
+	} else {
+		return nil, fmt.Errorf("found zero images based on regex %q and family %q in project %q", imageRegex, imageFamily, project)
```

different way of solving the silent failure issue.

---

Yes, I think about your change when refactoring the code here. Maybe we should put here since we expect at least one image found. No image means error, what do you think?

---

I agree if nobody is using the function anywhere else. I would need to check again and do another review.